developers/pymavlink.md

```diff
@@ -0,0 +1,245 @@
+
+```py
+# Import mavutil
+from pymavlink import mavutil
+
+# Create the connection
+# Need to provide the serial port and baudrate
+master = mavutil.mavlink_connection(
+            '/dev/ttyACM0',
+            baud=115200)
+
+# Restart the ArduSub board !
+master.reboot_autopilot()
+```
+
+##### ArduSub Board \(E.g: Pixhawk\) connected to the computer via UDP
+
+```py
+# Import mavutil
+from pymavlink import mavutil
+
+# Create the connection
+#  If using the companion computer
+#  the ip 192.168.2.1 and the port 14550
+#  will be available for connection
+master = mavutil.mavlink_connection('udp:192.168.2.1:14550')
+
+# Get some information !
+while True:
+    try:
+        print(master.recv_match().to_dict())
+    except:
+        pass
+'''
+Output:
+{'mavpackettype': 'AHRS2', 'roll': -0.11364290863275528, 'pitch': -0.02841472253203392, 'yaw': 2.0993032455444336, 'altitude': 0.0, 'lat': 0, 'lng': 0}
+{'mavpackettype': 'AHRS3', 'roll': 0.025831475853919983, 'pitch': 0.006112074479460716, 'yaw': 2.1514968872070312, 'altitude': 0.0, 'lat': 0, 'lng': 0, 'v1': 0.0, 'v2': 0.0, 'v3': 0.0, 'v4': 0.0}
+{'mavpackettype': 'VFR_HUD', 'airspeed': 0.0, 'groundspeed': 0.0, 'heading': 123, 'throttle': 0, 'alt': 3.129999876022339, 'climb': 3.2699999809265137}
+{'mavpackettype': 'AHRS', 'omegaIx': 0.0014122836291790009, 'omegaIy': -0.022567369043827057, 'omegaIz': 0.02394154854118824, 'accel_weight': 0.0, 'renorm_val': 0.0, 'error_rp': 0.08894175291061401, 'error_yaw': 0.0990816056728363}
+{'mavpackettype': 'HWSTATUS', 'Vcc': 4651, 'I2Cerr': 0}
+{'mavpackettype': 'SYSTEM_TIME', 'time_unix_usec': 0, 'time_boot_ms': 75191}
+{'mavpackettype': 'EKF_STATUS_REPORT', 'flags': 0, 'velocity_variance': 0.0, 'pos_horiz_variance': 0.000695356575306505, 'pos_vert_variance': 0.20162872970104218, 'compass_variance': 0.0037216085474938154, 'terrain_alt_variance': 0.04920071363449097}
+{'mavpackettype': 'VIBRATION', 'time_usec': 75191474, 'vibration_x': 0.03712763264775276, 'vibration_y': 0.03271860256791115, 'vibration_z': 0.05147671326994896, 'clipping_0': 0, 'clipping_1': 0, 'clipping_2': 0}
+{'mavpackettype': 'RAW_IMU', 'time_usec': 75430490, 'xacc': 6, 'yacc': -27, 'zacc': -1123, 'xgyro': -1, 'ygyro': 22, 'zgyro': -23, 'xmag': -353, 'ymag': -532, 'zmag': 257}
+{'mavpackettype': 'SCALED_IMU2', 'time_boot_ms': 75430, 'xacc': 39, 'yacc': 38, 'zacc': -980, 'xgyro': -45, 'ygyro': -65, 'zgyro': -13, 'xmag': 0, 'ymag': 0, 'zmag': 0}
+
+'''
+```
+
+##### Change mode
+
+```py
+# Import mavutil
+from pymavlink import mavutil
+
+# Create the connection
+master = mavutil.mavlink_connection('udp:192.168.2.1:14550')
+
+# Choose a mode
+mode = 'MANUAL'
+
+# Check if mode is available
+if mode not in master.mode_mapping():
+    print('Unknown mode : {}'.format(mode))
+    print('Try:', list(master.mode_mapping().keys()))
+    return
+
+# Get mode ID
+mode_id = master.mode_mapping()[mode]
+# Set new mode
+master.set_mode(mode_id)
```

Comment:
Show how to get the ack.
Also, let's show how to read current mode from heartbeat before and after changing it, use 'STABILIZE' because manual is default at boot.